File: ee25btech11047/Matgeo/problem21/codes/12.268.c
#include <stdio.h>

int main() {
    int trace = 11;
    int det = 36;
    int i, j, k;

    printf("The positive integer eigenvalues are:\n");

    for(i = 1; i <= trace; i++) {           // possible first eigenvalue
        for(j = 1; j <= trace - i; j++) {   // second eigenvalue
            k = trace - i - j;              // third eigenvalue from trace
            if(k > 0 && i * j * k == det) { // check product = determinant
                printf("Eigenvalues: %d, %d, %d\n", i, j, k);
                int max = i;
                if(j > max) max = j;
                if(k > max) max = k;
                printf("Largest eigenvalue: %d\n", max);
            }
        }
    }

    return 0;
}

File: ee25btech11047/Matgeo/problem21/codes/12.268.py
import numpy as np

# Inputs
trace = 11       # sum of eigenvalues
det = 36         # product of eigenvalues

# List to store valid eigenvalues
valid_eigenvalues = []

# Loop over all possible positive integers for first two eigenvalues
for i in range(1, trace + 1):
    for j in range(1, trace - i + 1):
        k = trace - i - j
        if k > 0 and i * j * k == det:
            eigenvalues = np.array([i, j, k])
            valid_eigenvalues.append(eigenvalues)
            print("Eigenvalues:", eigenvalues)
            print("Largest eigenvalue:", np.max(eigenvalues))

# If needed, store the largest eigenvalue among all triplets
if valid_eigenvalues:
    largest_overall = max(np.max(triplet) for triplet in valid_eigenvalues)
    print("Largest eigenvalue among all triplets:", largest_overall)
